pkg/raft/opts.go

```diff
@@ -126,6 +127,31 @@ func (this *options) ReadOption() raft.ReadOnlyOption {
 	return raft.ReadOnlySafe
 }
 
+func getLocalIPAddress() ([]net.IP, error) {
+	var ips []net.IP
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+	for _, i := range ifaces {
+		addrs, err := i.Addrs()
+		if err != nil {
+			return nil, err
+		}
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
```

Comment:
Given that there's identical processing inside both the case statements, check if the switch case is needed at all.

Reply:
Switch case is required, so that if addr is of any other type, that doesn't get added.